_storage/hub-operation.md

```diff
@@ -0,0 +1,108 @@
+---
+layout: storage
+permalink: /:collection/:path.html
+---
+# Understand hub operation
+{:.no_toc}
+
+This page describes the considerations hub operators must take into account when creating and operating a Gaia storage hub.
+
+* TOC
+{:toc}
+
+
+## Configuration files
+
+You should store a JSON configuration file either in the top-level directory of
+the hub server. Alternatively, you can specify a file location using the
+`CONFIG_PATH` environment variable.  The following is an example configuration file for Amazon S3:
+
+```json
+{
+  "servername": "localhost",
+  "port": 4000,
+  "driver": "aws",
+  "readURL": "https://YOUR_BUCKET_NAME.s3.amazonaws.com/",
+  "proofsConfig": {
+    "proofsRequired" : 0
+  },
+  "pageSize": 20,
+  "bucket": "YOUR_BUCKET_NAME",
+  "awsCredentials": {
+     "accessKeyID": "YOUR_ACCESS_KEY",
+     "secretAccessKey": "YOUR_SECRET_KEY"
+  },
+  "argsTransport": {
+    "level": "debug",
+    "handleExceptions": true,
+    "stringify": true,
+    "timestamp": true,
+    "colorize": false,
+    "json": true
+  }
+}
+```
+
+You can specify the logging level, the number of social proofs required
+for addresses to write to the system, the backend driver, the credentials
+for that backend driver, and the readURL for the storage provider.
+
+### Require the correct hub URL
+
+If you enable on the `requireCorrectHubUrl` option in your `config.json`
+file, your Gaia hub will require that authentication requests
+correctly include the `hubURL` they are trying to connect with. Use this option to prevent a malicious Gaia hub from using an authentication
+token for itself on other Gaia hubs.
+
+By default, the Gaia hub will validate that the supplied URL matches
+`https://${config.servername}`, but if there are multiple valid URLs
+for clients to reach the hub at, you can include a list in your `config.json`:
+
+```javascript
+{
+  ....
+  servername: "normalserver.com"
+  validHubUrls: [ "https://specialserver.com/",
+                  "https://legacyurl.info" ]
+  ....
+}
+```
+
+### The readURL parameter
+
+By default, hub drivers return read URLs that point directly at the written content. For example, an S3 driver would return the URL directly to the S3 file. If you configure a CDN or domain to point at that same bucket, you can use the `readURL` parameter to tell the hub that files can be read from a given URL. For example, the `hub.blockstack.org` Gaia Hub is configured to return a read URL that looks like `https://gaia.blockstack.org/hub/`.
+
+Unset the `readURL` parameter if you do not intend to deploy any caching.
+
+### The proofsRequired parameter
```

Comment:
Can we add a note here that this feature is deprecated? I'll add a note to the Gaia readme as well